### PR TITLE
Completed unit-tests-jest

### PR DESCRIPTION
**Pull request by techmatlock**

### CODE REVIEW

File: unit-tests-jest/numbers.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    expect(toDollars(5.86)).toEqual('$5.86');
+  });
+});
+
+describe('divideBy', () => {
+  it('divides by 4', () => {
+    expect(divideBy([2, 3, 4, 8], 4)).toEqual([0.5, 0.75, 1, 2]);
+  });
+  it('returns different array', () => {
+    expect(divideBy([2, 3, 4, 8], 4)).not.toBe([0.5, 0.75, 1, 2]);
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:38]: This doesn't actually test what you want it to. To check that, put the code back so there is a bug and you will see that this test still passes. The reason is that when you use a literal array expression (e.g., `[0.5, 0.75, 1, 2]`) it creates a _new_ instance. You need to assign the original array to a variable and then test that the result and the original are different.

File: unit-tests-jest/numbers.test.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        },
+        3
+      )
+    ).toEqual({
+      a: 3,
+      b: 6,
+      c: 9,
+    });
+  });
+  it('multiples by 2', () => {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:59]: I would have labeled this 'does not modify non-numbers'.

File: unit-tests-jest/numbers.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ */
+export function evenNumbers(numbers: number[]): number[] {
+  return numbers.filter((n) => n % 2 === 0);
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ */
+export function toDollars(amount: number): string {
+  if (amount < 0) return `-$${-amount}.00`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.ts:12]: Oops, this should use `toFixed` as well. I'm sure you can think of a test that will fail for this bug!

I do like you special handling of negative amounts, though.

**Review by RobertGardner — APPROVED** [review]

File: unit-tests-jest/numbers.test.ts (hunk 2 above)
Comment (techmatlock) [reply unit-tests-jest/numbers.test.ts:59]: Good catch. Thanks.

**Pull request merged**